code/implementation/system-stack/src/main/java/io/cattle/platform/systemstack/listener/SystemStackUpdate.java

```diff
@@ -69,6 +69,7 @@
     private static final String STACK_RESOURCE = "/config-content/system-stacks/%s/%s";
     private static final Logger log = LoggerFactory.getLogger(SystemStackUpdate.class);
     private static final String CATALOG_RESOURCE_URL = ArchaiusUtil.getString("system.stack.catalog.url").get();
+    private static final String CATALOG_RESOURCE_VERSION = ArchaiusUtil.getString("rancher.server.image").get();
```

Comment:
Can you not store the value as a final.  Can you convert both to being properly dynamic